lib/samson/process_utils.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Samson
+  module ProcessUtils
+    ATTRIBUTES = ['pid', 'ppid', 'gid', 'pcpu', 'user', 'start', 'args'].freeze
+
+    class << self
+      def ps_list
+        whitelists = ENV.fetch('PROCESS_WHITELIST', [])
+        pipe = IO.popen("ps -eo #{ATTRIBUTES.join(',')}")
+        # Filter the whitelist processes
+        filtered_processes = pipe.readlines[1..-1].reject do |lines|
+          whitelists.any? { |key| lines.include?(key) }
+        end
+        filtered_processes.map do |line|
+          Hash[ATTRIBUTES.zip line.lstrip.split(/\s+/, ATTRIBUTES.size)]
+        end
+      end
+
+      def report_to_statsd
+        ps_list.each do |process|
+          timed_out = Time.now.to_i - Time.parse(process.fetch('start')).to_i
+          tags = ATTRIBUTES.map { |attr| "#{attr}:#{process.fetch(attr)}" }
```

Comment:
can do:
`start = process[ATTRIBUTES.index('start')]`
and
`tags = ATTRIBUTES.each_with_index.map { |attr, i| "#{attr}:#{process[i]}" }`
to avoid hash coversion
... but :whatever 🤷‍♀️